gtests/net/packetdrill/run_packet.c

```diff
@@ -1248,14 +1248,18 @@ static int verify_outbound_live_tcp_options(
 	/* TCP options are expected to be a deterministic order. */
 	while (a_opt != NULL || s_opt != NULL) {
 		if (a_opt == NULL || s_opt == NULL ||
-		    a_opt->kind != s_opt->kind) {
+		    (a_opt->kind & ~TCPOPT_WILDCARD) !=
+		    (s_opt->kind & ~TCPOPT_WILDCARD)) {
 			asprintf(error, "bad outbound TCP options");
 			return STATUS_ERR;
 		}
 
-		if (verify_outbound_tcp_option(config, actual_packet,
-					script_packet, a_opt, s_opt,
-					error) != STATUS_OK) {
+		/* skip btye-to-byte comparison of wildcard option */
+		if ((a_opt->kind | s_opt->kind) & TCPOPT_WILDCARD)
```

Comment:
I think we only want to skip the verify_outbound_tcp_option() call if the *script* packet has the TCPOPT_WILDCARD bit set. If the actual packet has bit 0x80 set for some reason then AFAICT this indicates a kernel bug, and not something we should ignore. How about:

  if (s_opt->kind & TCPOPT_WILDCARD)

Reply:
There are options with the 0x80 bit set:
  https://www.iana.org/assignments/tcp-parameters/tcp-parameters.xhtml

So I would suggest tightening this up:

                /* Sometimes skip btye-to-byte comparison of wildcard option */
		if (s_opt->kind == (TCPOPT_WILDCARD | TCPOPT_WINDOW))